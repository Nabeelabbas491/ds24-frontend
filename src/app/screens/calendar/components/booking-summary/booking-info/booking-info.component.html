<div class="flex flex-row justify-start mt-5">
  <div class="w-12 h-12 bg-primary-p100 flex flex-col justify-center items-center" *ngIf="iconName">
    <ds-icon [name]="iconName" [color]="iconColors.Primary300" [size]="24" [title]="iconCaption"></ds-icon>
  </div>
  <div class="ml-3 flex flex-col flex-start">
    <h4 class="text-neutral-n500 font-medium text-left">{{ value }}</h4>
    <p class="text-sm mt-2 text-neutral-n300 text-left">{{ caption }}</p>
  </div>
</div>
