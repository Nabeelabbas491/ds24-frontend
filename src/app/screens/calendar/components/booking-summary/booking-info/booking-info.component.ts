import { Component, Input } from '@angular/core';
import { IconColor, IconName } from '@ds24/elements';

@Component({
  selector: 'ds-booking-info',
  templateUrl: './booking-info.component.html',
})
export class BookingInfoComponent {
  icons: typeof IconName = IconName;
  iconColors: typeof IconColor = IconColor;

  @Input()
  caption: string = '';

  @Input()
  value: string | null = '';

  @Input()
  iconName: IconName | undefined;

  @Input()
  iconCaption: string = '';
}
