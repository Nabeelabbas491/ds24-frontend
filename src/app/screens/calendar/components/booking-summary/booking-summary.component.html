<ds-booking-info
  [caption]="(selectCallTypeSummary$ | async)! | translate"
  [value]="(selectCallType$ | async)! | translate"
  [iconName]="(selectCallTypeIcon$ | async)!"
  [iconCaption]="'BOOKING.BOOKING_SUMMARY.BOOKING_TYPE_ICO' | translate"
></ds-booking-info>
<ds-booking-info
  [caption]="'BOOKING.BOOKING_SUMMARY.BOOKING_DATE_SUMMARY' | translate"
  [value]="(selectedDay$ | async)?.dayMonthYearTitle | date: 'fullDate'"
  [iconName]="icons.Calendar"
  [iconCaption]="'BOOKING.BOOKING_SUMMARY.BOOKING_DATE_ICO' | translate"
></ds-booking-info>
<ds-booking-info
  [caption]="'BOOKING.BOOKING_SUMMARY.BOOKING_TIME_SUMMARY' | translate"
  [value]="selectedTimeSlot$ | async | timeSlot"
  [iconName]="icons.Clock"
  [iconCaption]="'BOOKING.BOOKING_SUMMARY.BOOKING_TIMESLOT_ICO' | translate"
></ds-booking-info>
<div class="flex flex-row justify-center mt-5 mb-5" *ngIf="(isMobileView$ | async) === false">
  <button dsButton="default" (click)="editBooking()" [disabled]="(saveBookingPending$ | async)!">
    {{ 'BOOKING.BOOKING_OTHER.EDIT_BOOKING' | translate }}
  </button>
</div>
