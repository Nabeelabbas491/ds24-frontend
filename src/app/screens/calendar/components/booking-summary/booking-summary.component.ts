import { Component } from '@angular/core';
import { Store } from '@ngrx/store';
import { actions, selectors } from '../../../../store';
import { IconName } from '@ds24/elements';

@Component({
  selector: 'ds-booking-summary',
  templateUrl: './booking-summary.component.html',
})
export class BookingSummaryComponent {
  icons: typeof IconName = IconName;

  selectedDay$ = this.store.select(selectors.booking.selectSelectedDay);
  selectedTimeSlot$ = this.store.select(selectors.booking.selectTimeSlot);

  selectCallType$ = this.store.select(selectors.booking.selectCallType);
  selectCallTypeIcon$ = this.store.select(selectors.booking.selectCallTypeIcon);
  selectCallTypeSummary$ = this.store.select(selectors.booking.selectcallTypeSummary);

  isMobileView$ = this.store.select(selectors.viewBreakPoint.selectIsMobileView);

  saveBookingPending$ = this.store.select(selectors.bookingAPI.selectBookingSavePending);

  constructor(private store: Store) {}

  editBooking(): void {
    this.store.dispatch(actions.booking.editBooking());
  }
}
