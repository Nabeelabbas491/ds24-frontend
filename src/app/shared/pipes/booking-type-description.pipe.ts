import { Pipe, PipeTransform } from '@angular/core';
import { MeetingType, MeetingTypeDescriptionType } from './../../types/misc.type';
import { MeetingTypeId } from './../../types/booking.type';

@Pipe({
  name: 'bookingTypeDescription',
})
export class BookingTypeDescriptionPipe implements PipeTransform {
  transform(meetingType: MeetingType, descriptionType: MeetingTypeDescriptionType = 'default'): string {
    let description = '';
    if (descriptionType === 'default') {
      switch (meetingType.id) {
        case MeetingTypeId.Inbound:
          description = 'VENDOR.BOOKING.INBOUND_DESCRIPTION';
          break;
        case MeetingTypeId.Outbound:
          description = 'VENDOR.BOOKING.OUTBOUND_DESCRIPTION';
          break;
        case MeetingTypeId.Zoom:
          description = 'VENDOR.BOOKING.VIDEO_DESCRIPTION';
          break;
      }
    } else {
      switch (meetingType.id) {
        case MeetingTypeId.Inbound:
          description = 'VENDOR_SETTINGS.BOOKING_TYPE.INBOUND_PHONE_CALL.DESCRIPTION';
          break;
        case MeetingTypeId.Outbound:
          description = 'VENDOR_SETTINGS.BOOKING_TYPE.OUTBOUND_PHONE_CALL.DESCRIPTION';
          break;
        case MeetingTypeId.Zoom:
          description = 'VENDOR_SETTINGS.BOOKING_TYPE.VIDEO_CONFERENCE.DESCRIPTION';
          break;
      }
    }
    return description;
  }
}
