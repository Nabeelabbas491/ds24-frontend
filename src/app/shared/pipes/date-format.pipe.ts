import { Pipe, PipeTransform } from '@angular/core';
import { DateFormat } from './../../types/misc.type';
import { TranslateService } from '@ngx-translate/core';
import { format, parseISO } from 'date-fns';

@Pipe({
  name: 'dateFormat',
})
export class DateFormatPipe implements PipeTransform {
  constructor(private translateService: TranslateService) {}
  transform(dateString: string | undefined | null, formatStr: DateFormat = 'full'): string {
    if (dateString == null) {
      return '';
    }
    if (formatStr === 'full') {
      return format(parseISO(dateString), this.translateService.instant('CALENDAR.DAY_FORMAT_STRING.FULL'));
    }
    if (formatStr === 'full_without_year') {
      return format(
        parseISO(dateString),
        this.translateService.instant('CALENDAR.DAY_FORMAT_STRING.FULL_WITHOUT_YEAR'),
      );
    }

    if (formatStr === 'day_month_year') {
      return format(parseISO(dateString), 'dd MMMM, yyyy');
    }
    if (formatStr === 'day_half-month_year') {
      return format(parseISO(dateString), 'dd MMM, yyyy');
    }

    return '';
  }
}
