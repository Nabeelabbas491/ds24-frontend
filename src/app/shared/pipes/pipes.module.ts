import { NgModule } from '@angular/core';

import { AddCommasPipe } from './add-commas.pipe';
import { EllipsisPipe } from './ellipsis.pipe';
import { DateFormatPipe } from './date-format.pipe';
import { TimeSlotPipe } from './time-slot.pipe';
import { BookingTypeDescriptionPipe } from './booking-type-description.pipe';
import { TimezoneFormatPipe } from './timezone-format.pipe';
import { ShowPaginationPipe } from './show-pagination.pipe';

export const PIPES = [
  AddCommasPipe,
  EllipsisPipe,
  DateFormatPipe,
  TimeSlotPipe,
  BookingTypeDescriptionPipe,
  TimezoneFormatPipe,
  ShowPaginationPipe,
];

@NgModule({
  declarations: PIPES,
  exports: PIPES,
})
export class PipesModule {}
