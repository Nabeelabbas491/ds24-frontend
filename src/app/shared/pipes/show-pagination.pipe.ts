import { Pipe, PipeTransform } from '@angular/core';
import { PageBase } from 'src/app/types/misc.type';
import { showPagination } from '../common/util';

@Pipe({
  name: 'showPagination',
})
export class ShowPaginationPipe implements PipeTransform {
  transform(pageBase: PageBase): boolean {
    return showPagination(pageBase);
  }
}
