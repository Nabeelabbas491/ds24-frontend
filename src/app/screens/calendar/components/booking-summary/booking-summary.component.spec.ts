import { ComponentFixture, TestBed } from '@angular/core/testing';

import { BookingSummaryComponent } from './booking-summary.component';
import { TranslateModule } from '@ngx-translate/core';
import { IconModule } from '@ds24/elements';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { MockStore, provideMockStore } from '@ngrx/store/testing';
import { PipesModule } from '../../../../shared/pipes/pipes.module';
import { State, actions } from './../../../../store';
import { BookingInfoComponent } from './booking-info/booking-info.component';

describe('BookingSummaryComponent', () => {
  let fixture: ComponentFixture<BookingSummaryComponent>;
  let instance: BookingSummaryComponent;
  let mockStore: MockStore<State>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [BookingSummaryComponent, BookingInfoComponent],
      imports: [TranslateModule.forRoot(), IconModule, PipesModule, FormsModule, ReactiveFormsModule],
      providers: [provideMockStore()],
    });
    fixture = TestBed.createComponent(BookingSummaryComponent);
    instance = fixture.componentInstance;
    mockStore = TestBed.inject(MockStore);
    fixture.detectChanges();

    jest.spyOn(mockStore, 'dispatch');
  });

  it('should compile', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should dispatch edit booking action', () => {
    instance.editBooking();

    const action = actions.booking.editBooking();

    expect(mockStore.dispatch).toHaveBeenLastCalledWith(action);
  });

  it('should create', () => {
    expect(instance).toBeTruthy();
  });
});
