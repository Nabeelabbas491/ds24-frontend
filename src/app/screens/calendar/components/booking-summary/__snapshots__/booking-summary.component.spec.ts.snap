// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingSummaryComponent should compile 1`] = `
<ds-booking-summary
  icons={[Function Object]}
  isMobileView$={[Function _Store]}
  saveBookingPending$={[Function _Store]}
  selectCallType$={[Function _Store]}
  selectCallTypeIcon$={[Function _Store]}
  selectCallTypeSummary$={[Function _Store]}
  selectedDay$={[Function _Store]}
  selectedTimeSlot$={[Function _Store]}
  store={[Function _MockStore]}
>
  <ds-booking-info>
    <div
      class="flex flex-row justify-start mt-5"
    >
      <div
        class="ml-3 flex flex-col flex-start"
      >
        <h4
          class="text-neutral-n500 font-medium text-left"
        >
        </h4>
        <p
          class="text-sm mt-2 text-neutral-n300 text-left"
        >
        </p>
      </div>
    </div>
  </ds-booking-info><ds-booking-info>
    <div
      class="flex flex-row justify-start mt-5"
    >
      <div
        class="w-12 h-12 bg-primary-p100 flex flex-col justify-center items-center"
      >
        <ds-icon
          title="BOOKING.BOOKING_SUMMARY.BOOKING_DATE_ICO"
        >
          <calendar-icon>
            <svg
              fill="none"
              height="24"
              style="height: 24px; width: 24px;"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="fill-primary-p300-main"
                clip-rule="evenodd"
                d="M17.3853 3V4.54101H22V21.227H2V4.54101H6.61475V3H8.15576V4.54101H15.8442V3H17.3853ZM3.54101 19.6943H20.459V6.07374H17.3853V7.61475H15.8525V6.07374H8.15576V7.61475H6.61475V6.07374H3.54101V19.6943ZM8.15576 11.1691H6.61475V12.7101H8.15576V11.1691ZM11.2295 11.1691H9.68848V12.7101H11.2295V11.1691ZM12.7705 11.1691H14.3115V12.7101H12.7705V11.1691ZM17.3853 14.2428H15.8442V15.7838H17.3853V14.2428ZM6.61475 14.2428H8.15576V15.7838H6.61475V14.2428ZM11.2295 14.2428H9.68848V15.7838H11.2295V14.2428ZM12.7705 14.2428H14.3115V15.7838H12.7705V14.2428ZM17.3853 11.1691H15.8442V12.7101H17.3853V11.1691Z"
                fill-rule="evenodd"
              />
            </svg>
          </calendar-icon>
        </ds-icon>
      </div>
      <div
        class="ml-3 flex flex-col flex-start"
      >
        <h4
          class="text-neutral-n500 font-medium text-left"
        >
        </h4>
        <p
          class="text-sm mt-2 text-neutral-n300 text-left"
        >
          BOOKING.BOOKING_SUMMARY.BOOKING_DATE_SUMMARY
        </p>
      </div>
    </div>
  </ds-booking-info><ds-booking-info>
    <div
      class="flex flex-row justify-start mt-5"
    >
      <div
        class="w-12 h-12 bg-primary-p100 flex flex-col justify-center items-center"
      >
        <ds-icon
          title="BOOKING.BOOKING_SUMMARY.BOOKING_TIMESLOT_ICO"
        >
          <clock-icon>
            <svg
              fill="none"
              style="height: 24px; width: 24px;"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="fill-primary-p300-main"
                clip-rule="evenodd"
                d="M0 10C0 4.47715 4.47715 0 10 0C12.6522 0 15.1957 1.05357 17.0711 2.92893C18.9464 4.8043 20 7.34784 20 10C20 15.5228 15.5228 20 10 20C4.47715 20 0 15.5228 0 10ZM16.0763 12.376V10.376H11V4H9V12.376H16.0763Z"
                fill-rule="evenodd"
                id="Mask"
              />
            </svg>
          </clock-icon>
        </ds-icon>
      </div>
      <div
        class="ml-3 flex flex-col flex-start"
      >
        <h4
          class="text-neutral-n500 font-medium text-left"
        >
        </h4>
        <p
          class="text-sm mt-2 text-neutral-n300 text-left"
        >
          BOOKING.BOOKING_SUMMARY.BOOKING_TIME_SUMMARY
        </p>
      </div>
    </div>
  </ds-booking-info>
</ds-booking-summary>
`;
